egui_extras/src/table.rs

```diff
@@ -351,7 +351,65 @@ pub struct TableBody<'a> {
     end_y: f32,
 }
 
+pub trait TableRowBuilder {
+    fn row_heights(&self, widths: &Vec<f32>) -> Box<dyn Iterator<Item = f32> + '_>;
```

Comment:
note for reviewers: widths isn't used in the demo app, but I included it here because my main use case for the `Table` type is to implement a chat box where text will be wrapped based on width and so height will be tied to cell width.

actually, come to think of it this was the main reason i thought it would be good to use a trait rather than simply taking an iterator over `f32` without any input from `heterogenous_rows`.

Reply:
if there were a `widths(&self) -> Vec<32>` method on `TableBody` then coming up with an iterator of floating point height values wouldn't explicitly require it as an input

Reply:
I agree that using a trait for this is quite cumbersome. It would be much nicer to use with just closures.

Adding `fn width(&self) -> &[f32]` to `TableBody` is easy enough.